File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/BulletHitBulletEvent.java
package dev.robocode.tankroyale.botapi.events;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;
import dev.robocode.tankroyale.botapi.BulletState;

/** Event occurring when a bullet has collided with another bullet */
@Value
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public final class BulletHitBulletEvent extends Event {
  /** Bullet that hit another bullet */
  BulletState bullet;
  /** The other bullet that was hit by the bullet */
  BulletState hitBullet;

  @Builder
  @SuppressWarnings("UnusedDeclaration")
  private BulletHitBulletEvent(int turnNumber, BulletState bullet, BulletState hitBullet) {
    this.turnNumber = turnNumber;
    this.bullet = bullet;
    this.hitBullet = hitBullet;
  }
}

File: bot-api/dotnet/Robocode.TankRoyale.BotApi/src/events/BulletHitBulletEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when a bullet has collided with another bullet.
/// </summary>
public sealed class BulletHitBulletEvent : BotEvent
{
    /// <summary>Bullet that hit another bullet.</summary>
    public BulletState Bullet { get; }

    /// <summary>The other bullet that was hit by the bullet.</summary>
    public BulletState HitBullet { get; }

    /// <summary>
    /// Initializes a new instance of the BulletHitBulletEvent class.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that hit another bullet.</param>
    /// <param name="hitBullet">The other bullet that was hit by the bullet.</param>
    [JsonConstructor]
    public BulletHitBulletEvent(int turnNumber, BulletState bullet, BulletState hitBullet) : base(turnNumber) =>
        (Bullet, HitBullet) = (bullet, hitBullet);
}